CK3toEU4/Source/EU4World/FlagFoundry/FlagFoundry.cpp

```diff
@@ -97,20 +103,56 @@ void EU4::FlagFoundry::craftFlag(const std::shared_ptr<Country>& country) const
 	}
 }
 
+void EU4::FlagFoundry::craftRebelFlag(const Configuration& theConfiguration, const GeneratedReligion& religion) const
+{
+	//Import the generic Rebel Flag	
+	if (!commonItems::DoesFileExist("blankMod/output/gfx/flags/generic_rebels.tga"))
+		throw std::runtime_error("blankMod/output/gfx/flags/generic_rebels.tga! Where are the rebel scum!?");
+	Magick::Image baseFlag("blankMod/output/gfx/flags/generic_rebels.tga");
+	
+	//Process target icon
+	Magick::Image targetIcon;
+	if (!religion.iconPath.empty() && commonItems::DoesFileExist(theConfiguration.getCK3Path() + "gfx/interface/icons/religion/" + religion.iconPath + ".dds"))
+	{
+		targetIcon.read(theConfiguration.getCK3Path() + "gfx/interface/icons/religion/" + religion.iconPath + ".dds");
+		Log(LogLevel::Debug) << religion.name << " icon path: " << religion.iconPath;
+	}
+	else
+	{
+		Log(LogLevel::Warning) << "Could not find " << theConfiguration.getCK3Path() + "gfx/interface/icons/religion/" + religion.iconPath << ".dds, skipping!";
+		// blank.
+		targetIcon = Magick::Image("100x100", Magick::Color("transparent"));
+	}
+
+	//Turn the Icon pure white
+	targetIcon.gamma(255);
+
+	//Now make the Icon a little smaller and flip it right-side up （85/85 instead of 100/100)
+	targetIcon.adaptiveResize(Magick::Geometry(targetIcon.size().width() * 0.85, targetIcon.size().height() * 0.85));
+	targetIcon.flip();
+
+	//Finally, combine the images into a new one
+	baseFlag.composite(targetIcon, MagickCore::CenterGravity, MagickCore::OverCompositeOp);
+	
+	//Output
+	baseFlag.write("flags.tmp/" + religion.name + "_rebels.tga");
+
+}
+
 void EU4::FlagFoundry::extendReligionStrips(const Configuration& theConfiguration, const std::vector<GeneratedReligion>& religions) const
 {
 	std::set<std::string> targetStrips = {"country_icon_religion.dds", "icon_religion.dds", "icon_religion_small.dds", "province_view_religion.dds"};
 	for (const auto& religion: religions)
 	{
 		// Import the source dds
 		Magick::Image sourceIcon;
-		if (!religion.iconPath.empty() && commonItems::DoesFileExist(theConfiguration.getCK3Path() + religion.iconPath))
+		if (!religion.iconPath.empty() && commonItems::DoesFileExist(theConfiguration.getCK3Path() + "gfx/interface/icons/religion/" + religion.iconPath + ".dds"))
```

Comment:
these changes are out of sync with current code. Custom religions store their icons with absolute paths.